### PR TITLE
docs: lua2dox.lua debugging, fix vim.treesitter tags

### PR DESCRIPTION
**Pull request by justinmk**
## Problem:

1. Help tags like vim.treesitter.language.add() are confusing because `vim.treesitter.language` is (thankfully) not a user-facing module.
2. Not easy to debug `lua2dox.lua` script.

## Solution:

1. Ignore the "fstem" when generating "treesitter" tags.
2. Let `gen_vimdoc.py ... --keep-tmpfiles` write the lua2dox filtered quasi-C output for inspection.

### CODE REVIEW

File: runtime/lua/vim/treesitter/languagetree.lua [hunk 1]
```diff
@@ -38,6 +38,8 @@ local LanguageTree = {}
 
 LanguageTree.__index = LanguageTree
 
+--- @private
+---
 --- A |LanguageTree| holds the treesitter parser for a given language {lang} used
```
Comment (justinmk) [inline runtime/lua/vim/treesitter/languagetree.lua:43]: It seems like `LanguageTree.new` and `TSHighlighter.new` were being used as "module-level docstrings". Do we not have something like `\defgroup` for Lua modules? https://github.com/neovim/neovim/blob/6d4f48182131c36d57589eefd4cefe3c70256d04/src/nvim/api/buffer.c#L47

**Comment by lewis6991** [comment]
I don't really like this change as it immediately implies `vim.treesitter.add()` is callable as `lua vim.treesitter.add()` (which it isn't). If the tags are annoying, then I'd rather omit them completely.

**Comment by justinmk** [comment]
> this change as it implies `vim.treesitter.add()` is callable as `lua vim.treesitter.add()` (which it isn't).

? It is callable:

    lua vim.treesitter.add('help', {})

and it certainly isn't callable as `vim.treesitter.language.add()`, which is the current tag.

**Comment by lewis6991** [comment]
Ah, I forgot about that metatable. Sorry for the noise. This is good.

File: scripts/lua2dox.lua [hunk 2]
```diff
@@ -24,11 +24,18 @@ Lua-to-Doxygen converter
...
 -------
 
 This script "lua2dox.lua" gets called by "gen_vimdoc.py".
 
+DEBUGGING/DEVELOPING
+---------------------
+
+1. To debug, run gen_vimdoc.py with --keep-tmpfiles:
+   python3 scripts/gen_vimdoc.py -t treesitter --keep-tmpfiles
+2. The filtered result will be written to ./tmp-lua2dox-doc/….lua.c
```
Comment (justinmk) [inline scripts/lua2dox.lua:37]: Now `gen_vimdoc.py ... --keep-tmpfiles` will write the lua2dox filtered quasi-C output for inspection.

(How were people debugging this before?)

Comment (clason) [reply scripts/lua2dox.lua:37]: Trial and error, of course ;)

File: runtime/lua/vim/treesitter/languagetree.lua (hunk 1 above)
Comment (justinmk) [reply runtime/lua/vim/treesitter/languagetree.lua]: Revisiting this in https://github.com/neovim/neovim/pull/22498

**Pull request merged**